src/core/lib/transport/interception_chain.h

```diff
@@ -114,6 +114,10 @@ class Interceptor : public UnstartedCallDestination {
                });
   }
 
+  // Hijack a call with custom initial metadata.
+  CallInitiator MakeChildCall(ClientMetadataHandle metadata,
```

Comment:
It seems sub-optimal to have two different APIs for hijacked calls.  Why do we need a different API than the existing one?  Do we expect that there are actually two different use-cases here, and we need to provide one API optimized for each one?  Can we somehow coalesce these into a single API?

If we do keep this, then I think we need more documentation to explain how it should be used and how an interceptor would choose between these two hijacking APIs.

Reply:
Yeah Hijack didn't work out for this use case at all... metadata availability was in the wrong place and it was a mess.
I want to check a few other use cases before removing it, but I think I want to be removing it.
Left a comment with more detail.